cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/BasicSparkClientContext.java

```diff
@@ -159,6 +160,11 @@ public Admin getAdmin() {
   }
 
   @Override
+  public DataTracer getDataTracer(String dataTracerName) {
```

Comment:
Is the `DataTracer` returned serializable? If not, then it can't be used inside spark function through closure.